generator/generator.go

```diff
@@ -540,14 +543,25 @@ func (g *generator) writeNode(indentLevel int, current parser.Node, next parser.
 	case parser.RawElement:
 		err = g.writeRawElement(indentLevel, n)
 	case parser.ForExpression:
+		maybeWhitespace = false
 		err = g.writeForExpression(indentLevel, n, next)
 	case parser.CallTemplateExpression:
 		err = g.writeCallTemplateExpression(indentLevel, n)
 	case parser.TemplElementExpression:
 		err = g.writeTemplElementExpression(indentLevel, n)
+
+		// TemplElementExpression with block should always have whitespace if the next element is also
```

Comment:
Great feedback! Let me explain:

The full comment is on that line is:
```
		// TemplElementExpression with block should always have whitespace if the next element is also
		// a TemplElementExpression
```
I wrapped it to keep within a sensible line width.

Templ elements can contain children, which is a block, and is called as such in the code base in tests. I merely used the same terminology.
```
		{
			name: "templelement: simple, block with text",
			input: `@Other(p.Test) {
	some words
}`,
```

When I didn't add this special case, tests were failing. See my comment in one of the outdated threads.
There are tests that expect horizontal spacing between 2 TemplElementExpressions that contain children. Since I did not want to change the test, in order to maintain the original functionality, I added this edge-case, as it was the only way I could figure out to satisfy the existing test.
See test `test-whitespace-around-go-keywords`, which includes multiple Templ elements.
I don't think adding a test for this is necessary since this test already does exactly this, albeit with a dumb name.

The nil check was my bad, missed that one, sorry.

Reply:
> if the comment is accurate, with <i>child text node</i><b>next node</b>, we might end up with a line break between those inline nodes, since whitespace would be "forced".

Let me clarify that what it's adding here is a space, not a newline. There are tests that check for this.